### PR TITLE
feat: warn if docstring has extra parameters

### PR DESCRIPTION
**Pull request by KevinMusgrave**
Reference: https://github.com/mkdocstrings/mkdocstrings/issues/408

### CODE REVIEW

File: src/griffe/dataclasses.py [hunk 1]
```diff
@@ -247,6 +247,9 @@ def __len__(self):
     def __iter__(self):
         return iter(self._parameters_list)
 
+    def __contains__(self, item):
+        return item in self._parameters_dict
```
Comment (pawamoy) [inline src/griffe/dataclasses.py:251]:
```suggestion
    def __contains__(self, param_name: str):
        return param_name in self._parameters_dict
```

File: src/griffe/docstrings/google.py [hunk 2]
```diff
@@ -202,6 +204,14 @@ def _read_parameters(docstring: Docstring, offset: int) -> tuple[list[DocstringP
         if annotation is None:
             _warn(docstring, line_number, f"No type or annotation for parameter '{name}'")
 
+        if warn_unknown_params and docstring.parent is not None:
+            if name not in docstring.parent.parameters:
+                _warn(
+                    docstring,
+                    line_number,
+                    f'The docstring for {docstring.parent} contains "{name}" as a parameter, but "{name}" is not an argument.',
```
Comment (pawamoy) [inline src/griffe/docstrings/google.py:212]:
```suggestion
                    f"Parameter '{name}' does not appear in the parent signature",
```

File: tests/test_docstrings/test_google.py [hunk 3]
```diff
@@ -702,22 +702,34 @@ def test_parameter_line_without_colon_keyword_only(parse_google):
...
+                Parameter("y"),
+            ),
+        ),
+    )
+    assert len(sections) == 1
+    assert len(warnings) == 3
+    assert (
+        'The docstring for <Function(\'func\', None, None)> contains "x" as a parameter, but "x" is not an argument.'
+        in warnings
+    )
```
Comment (pawamoy) [inline tests/test_docstrings/test_google.py:732]:
```suggestion
    assert "'x' does not appear in parent signature" in warnings[1]
```

**Review by pawamoy — CHANGES_REQUESTED** [review]
Hey, thanks a lot!

A few suggestions. We also must pass `warn_unknown_params=False` from `_read_other_parameters`, and add a test to make sure "Other Parameters" sections will not emit warnings.

**Comment by KevinMusgrave** [comment]
I added ```warn_unknown_params=False``` and tested for it in ```test_extra_parameter```

**Comment by pawamoy** [comment]
I've split the test in two, to make it more resilient to future possible changes in the warning string: we just check no warnings were emitted.

**Review by pawamoy — APPROVED** [review]
LGTM

**Comment by pawamoy** [comment]
Thanks again!

**Pull request merged**